synapse/lib/node.py

```diff
@@ -222,10 +222,15 @@ def getNodeRefs(self):
             return False
 
         if self.isrunt:
+
             if prop.info.get('ro'):
-                raise s_exc.IsRuntForm(mesg='Cannot set read-only props on runt nodes',
-                                       form=self.form.full, prop=name, valu=valu)
-            return await self.snap.core.runRuntPropSet(self, prop, valu)
+                mesg = 'Cannot set read-only props on runt nodes'
+                raise s_exc.IsRuntForm(mesg=mesg, form=self.form.full, prop=name, valu=valu)
+
+            await self.snap.core.runRuntPropSet(self, prop, valu)
+            self.props[prop.name] = valu
```

Comment:
`runRuntPropSet` should be responsible for modifying the `Node` object directly. We shouldn't do anything with self.props here.